napari/components/tests/test_viewer_model.py

```diff
@@ -315,3 +317,54 @@ def test_add_remove_layer_dims_change():
     viewer.layers.remove(layer)
     assert len(viewer.layers) == 0
     assert viewer.dims.ndim == 2
+
+
+good_layer_data = [
+    (np.random.random((10, 10)),),
+    (np.random.random((10, 10, 3)), {'rgb': True}),
+    (np.random.randint(20, size=(10, 15)), {'seed': 0.3}, 'labels'),
+    (np.random.random((10, 2)) * 20, {'face_color': 'blue'}, 'points'),
+    (np.random.random((10, 2, 2)) * 20, {}, 'vectors'),
+    (np.random.random((10, 4, 2)) * 20, {'opacity': 1}, 'shapes'),
+    (
+        (
+            np.random.random((10, 3)),
+            np.random.randint(10, size=(6, 3)),
+            np.random.random(10),
+        ),
+        {'name': 'some surface'},
+        'surface',
+    ),
+]
+
+
+@pytest.mark.parametrize('data', good_layer_data)
+def test_add_layer_data(data):
+    # make sure adding valid layer data calls the proper corresponding add_*
+    # method for all layer types
+    viewer = ViewerModel()
+    viewer._add_layer_data(*data)
+    assert len(viewer.layers) == 1
```

Comment:
can you clarify?  do you mean that the `data` format matches the `layer_type`? or that the `layer_type` is a valid string?

Reply:
ohhh... sorry, I get it.  that the layer that got added to the viewer is the right type, right?

Reply:
Ah sorry for being unclear. Maybe it's overkill, but I was thinking it would be nice to make sure the default `layer_type` and method name lookup are working. Something like:

```python
assert isinstance(viewer.layers[0], correct_layer_type)
```